api/cmd/api/ui_handler.go

```diff
@@ -11,19 +11,10 @@ import (
 )
 
 type uiEnvHandler struct {
-	OauthClientID     string                `json:"REACT_APP_OAUTH_CLIENT_ID,omitempty"`
-	Environment       string                `json:"REACT_APP_ENVIRONMENT,omitempty"`
-	SentryDSN         string                `json:"REACT_APP_SENTRY_DSN,omitempty"`
-	DocURL            config.Documentations `json:"REACT_APP_MERLIN_DOCS_URL,omitempty"`
-	HomePage          string                `json:"REACT_APP_HOMEPAGE,omitempty"`
-	MerlinURL         string                `json:"REACT_APP_MERLIN_API,omitempty"`
-	MlpURL            string                `json:"REACT_APP_MLP_API,omitempty"`
-	DockerRegistries  string                `json:"REACT_APP_DOCKER_REGISTRIES,omitempty"`
-	MaxAllowedReplica int                   `json:"REACT_APP_MAX_ALLOWED_REPLICA,omitempty"`
+	*config.ReactAppConfig
```

Comment:
What's the criteria for a config to be within `ReactAppConfig` or `uiEnvHandler` ?

Reply:
`ReactAppConfig` holds config values, that are populated directly via `REACT_APP_*` env vars configured on the `merlin` container. 

`uiEnvHandler` – leverages config from `ReactAppConfig` and adds on top of it cross-component config values (i.e. feature flags): https://github.com/gojek/merlin/blob/41b648d61c1dd151dc02d88a317e7b90a39af3ee/api/cmd/api/main.go#L112

Reply:
I think I get it. So config in `uiEnvHandler` has `REACT_APP_*` json representation for writing, not for reading?
That's confusing if true @ariefrahmansyah

Reply:
> So config in uiEnvHandler has `REACT_APP_*` json representation for writing, not for reading?

This is accurate. Generally, I'd like us to spend more time addressing the maintainability of our existing codebases because they expanded substantially over time and now new changes added to the repo are addressing a local problem (i.e. how to make this specific function work correctly) without taking into account the big picture of the entire repo. 

Example: 
These two environment variables have no effect on the deployment at all:
https://github.com/gojek/merlin/blob/28d764b4fc84009c78aadddda0962c415e6f4d5b/charts/merlin/templates/merlin-deployment.yaml#L97-L100

Reply:
> So config in `uiEnvHandler` has `REACT_APP_*` json representation for writing, not for reading?
Do you mean `uiEnvHandler` writes `REACT_APP_*` and additional config values as the response when the user visits `/env.js`? If it's true, then I get you, otherwise, can you elaborate? @pradithya 

> I'd like us to spend more time addressing the maintainability of our existing codebases
Agree. One of the steps that I would like to take is to refactor the deployment configuration so it can have a better locality and still improve reusability. We are planning to achieve this in this semester 🤞🏻

Reply:
> I would like to take is to refactor the deployment configuration

As part of this, it would be great to move most of the configuration from env vars to a config file, and only keep the sensitive overrides in env vars.